go-controller/pkg/util/pod_annotation.go

```diff
@@ -458,8 +458,36 @@ func AddRoutesGatewayIP(
 		podAnnotation.Gateways = append(podAnnotation.Gateways, network.GatewayRequest...)
 		topoType := netinfo.TopologyType()
 		switch topoType {
-		case types.Layer2Topology, types.LocalnetTopology:
+		case types.Layer2Topology:
 			// no route needed for directly connected subnets
+			return nil
+		case types.LocalnetTopology:
+			// no route needed for directly connected subnets
```

Comment:
but shouldn't we need to configure the GW if one is provided ?

Reply:
here is a pod whose second interface is the localnet network in my setup:

<localnet_cidr> dev net1 scope link  src <localnet_interface_ip>

so you don't need any other route to access localnet_cidr

Reply:
By GW I meant default route. Sorry for not being clear :)

Reply:
I looked in more detail, and now see I failed to notice the GW is carried to the annotation a few lines above (line 458).